app/site-watcher.ts

```diff
@@ -143,6 +144,12 @@ export async function watchSites(
       if (!json.name) {
         json.name = packageJson.name
       }
+
+      if (!json.hash) {
+        const [hash] = metadataPath.split(`/`)
+        json.hash = hash
+      }
```

Comment:
not sure I understand what is going on here... maybe a code comment to explain?

Reply:
Will do. The metadataPath is something like `"314a8e6cf402dfc76d38cf99c535a4a2/metadata.json"`. This just grabs the first item in the path, which is the site hash